toolkits/jira/arcade_jira/utils.py

```diff
@@ -1222,9 +1223,53 @@ def extract_id(field: Any) -> dict[str, str] | None:
 
 
 async def resolve_cloud_id(context: ToolContext, cloud_id: str | None) -> str:
+    try:
+        uuid.UUID(cloud_id)
+    except (AttributeError, TypeError, ValueError):
+        is_valid_uuid = False
+    else:
+        is_valid_uuid = True
+
+    # If this is already a valid Cloud ID, return it
+    if is_valid_uuid:
+        return cast(str, cloud_id)
+
+    # If not, it's possibly a Cloud name, so we try to match that.
     if isinstance(cloud_id, str) and cloud_id != "":
-        return cloud_id
+        return await get_cloud_id_by_cloud_name(context, cloud_name=cloud_id)
+
+    # As a last resort, try to get a unique Cloud ID from the available Atlassian Clouds
+    return await get_unique_cloud_id(context)
+
+
+async def get_cloud_id_by_cloud_name(context: ToolContext, cloud_name: str) -> str:
+    from arcade_jira.tools.cloud import get_available_atlassian_clouds  # Avoid circular import
```

Comment:
little code smell here. usually means restructure unless it's third party dep you can't control.

Reply:
True, we could have a separate module with routines that only depend on tools. Will work on this clean up.